Tests/ServiceLifecycleTests/GracefulShutdownTests.swift

```diff
@@ -253,4 +253,23 @@ final class GracefulShutdownTests: XCTestCase {
             XCTAssertTrue(Task.isShuttingDownGracefully)
         }
     }
+    
+    func testWaitForGracefulShutdown() async throws {
```

Comment:
Can we have two more tests here:
1. Where graceful shutdown is already triggered before calling `await gracefulShutdown()`
2. Where we cancel while awaiting the graceful shutdown